all/test.sh

```diff
@@ -375,9 +381,29 @@ for lang in ${LANGS[@]}; do
     fi
 
     # Test without an output directory.
-    testGeneration "$lang" "$lang" "$expected_output_dir" 0
+    testGeneration "$lang" "$lang" "false" "$expected_output_dir" 0
 
     # Test with an output directory.
     test_dir="gen/foo/bar"
-    testGeneration "${lang}_with_output_dir" "$lang" "$test_dir" 0 -o "$test_dir"
+    testGeneration "${lang}_with_output_dir" "$lang" "false" "$test_dir" 0 -o "$test_dir"
 done
+
+#Generate proto files but using the -d switch to just pass a directory input
+for lang in ${LANGS[@]}; do
```

Comment:
mind refactoring this a little bit and use the same loop to test all `withdir` true/false cases?